python/cugraph/dask/community/louvain.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dask.distributed import wait, default_client
+
+import cugraph.comms.comms as Comms
+from cugraph.dask.common.input_utils import get_local_data
+
+from cugraph.dask.community import louvain_wrapper as c_mg_louvain
+
+
+def call_louvain(sID, data, local_data, max_level, resolution):
+    wid = Comms.get_worker_id(sID)
+    handle = Comms.get_handle(sID)
+    return c_mg_louvain.louvain(data[0],
+                                local_data,
+                                wid,
+                                handle,
+                                max_level,
+                                resolution)
+
+
+def louvain(input_graph, max_iter=100, resolution=1.0, load_balance=True):
+    """
+    Compute the modularity optimizing partition of the input graph using the
+    Louvain method on multiple GPUs
+
+    Examples
+    --------
+    >>> import cugraph.dask as dcg
+    >>> Comms.initialize()
+    >>> chunksize = dcg.get_chunksize(input_data_path)
+    >>> ddf = dask_cudf.read_csv('datasets/karate.csv', chunksize=chunksize,
+                                 delimiter=' ',
+                                 names=['src', 'dst', 'value'],
+                                 dtype=['int32', 'int32', 'float32'])
+    >>> dg = cugraph.Graph()
+    >>> dg.from_dask_cudf_edgelist(ddf, source='src', destination='dst',
+                                   edge_attr='value')
+    >>> parts, modularity_score = dcg.louvain(dg)
+    """
+    # FIXME: finish docstring: describe parameters, etc.
+
+    # FIXME: import here to prevent circular import: cugraph->louvain
+    # wrapper->cugraph/structure->cugraph/dask->dask/louvain->cugraph/structure
+    # from cugraph.structure.graph import Graph
+
+    # FIXME: dask methods to populate graphs from edgelists are only present on
+    # DiGraph classes. Disable the Graph check for now and assume inputs are
```

Comment:
Why are we exposing symmetric DiGraph and throw errors for `Graph` or actual `DiGraph`? It is a bit awkward from a UX perspective since cuGraph has the `Graph` class for undirected graphs.

Reply:
Agreed, there's still some tech debt there that I may have to address in the next PR.